pkg/windows/calico.go

```diff
@@ -52,147 +53,226 @@ const (
+			LogSeverityFile: "none",
+			LogSeveritySys:  "none",
+		},
+		CNI: CalicoCNIConfig{
+			BinDir:   nodeConfig.AgentConfig.CNIBinDir,
+			ConfDir:  nodeConfig.AgentConfig.CNIConfDir,
+			IpamType: "calico-ipam",
+			Version:  "0.3.1",
+		},
 	}
 
-	cfg.CalicoConfig.KubeConfig = calicoKubeConfig
-	if err := getCNIConfigOverrides(cfg, hc); err != nil {
-		return nil, err
+	c.CalicoCNICfg.CalicoConfig.KubeConfig, err = c.createCalicoKubeConfig(ctx, restConfig)
+	if err != nil {
+		return err
 	}
 
-	agent.NetworkName = CalicoHnsNetworkName
+	return nil
+}
 
-	if err := createNodeNameFile(cfg, dataDir); err != nil {
-		return nil, err
+// writeConfigFiles writes the three required files by Calico
+func (c *Calico) writeConfigFiles(CNIConfDir string, NodeName string) error {
+	// parse the two config files required by Calico
+	templates, err := parseCalicoTemplatesFromConfig(c.CalicoCNICfg)
+	if err != nil {
+		return err
+	}
+	
+	// create calico kubeconfig, calico config and node name files
+	if err = util.WriteFile(filepath.Join("c:\\", c.DataDir, "agent", CalicoKubeConfigName), templates[calicoKubeConfigTemplate]); err != nil {
+		return err
+	}
+	if err = util.WriteFile(filepath.Join(CNIConfDir, CalicoConfigName), templates[calicoConfigTemplate]); err != nil {
+		return err
+	}
+    if err = util.WriteFile(c.CalicoCNICfg.CalicoConfig.KubeConfig.Path, NodeName); err != nil {
+		return err
 	}
-	if err := createKubeConfig(cfg, dataDir); err != nil {
-		return nil, err
+
+	return nil
+}
+
+
+// createCalicoKubeConfig creates all needed for Calico to contact kube-api
+func (c *Calico) createCalicoKubeConfig(ctx context.Context, restConfig *rest.Config) (CalicoKubeConfig, error) {
+
+	// Fill all information except for the token
+	calicoKubeConfig := CalicoKubeConfig{
+		Server:               "https://127.0.0.1:6443",
+		CertificateAuthority: filepath.Join("c:\\", c.DataDir, "agent", "server-ca.crt"),
+		Token:                "",
+		Path:                 filepath.Join("c:\\", c.DataDir, "agent", CalicoKubeConfigName),
 	}
 
-	if err := createCNIConfig(cfg); err != nil {
-		return nil, err
+	// Generate the token request
+	req := authv1.TokenRequest{
+		Spec: authv1.TokenRequestSpec{
+			Audiences:         []string{version.Program},
+			ExpirationSeconds: pointer.Int64(60 * 60 * 24 * 365),
+		},
 	}
 
-	logrus.Info("Generating HNS networks, please wait")
-	if err := generateCalicoNetworks(cfg.CalicoConfig.NetworkingBackend); err != nil {
-		return nil, err
+	// Register the token in the Calico service account
+	client, err := kubernetes.NewForConfig(restConfig)
+	if err != nil {
+		return calicoKubeConfig, err
+	}
+	serviceAccounts := client.CoreV1().ServiceAccounts(CalicoSystemNamespace)
+	token, err := serviceAccounts.CreateToken(ctx, calicoNode, &req, metav1.CreateOptions{})
+	if err != nil {
+		return calicoKubeConfig, errors.Wrapf(err, "failed to create token for service account (%s/%s)", CalicoSystemNamespace, calicoNode)
```

Comment:
same comment here - return nil pointer instead of unusable struct
```suggestion
		return nil, errors.Wrapf(err, "failed to create token for service account (%s/%s)", CalicoSystemNamespace, calicoNode)
```

Reply:
yes